### DISCUSSION TITLE
ERROR: node:internal/process/esm_loader:46 internalBinding('errors').triggerUncaughtException

### DISCUSSION THREAD

**Discussion by bustahxc** [webdriverio/webdriverio#10850] [Q&A] [answered]
I created a new WebdriverIO project with TypeScript and Cucumber, and when trying to run a test, it throws the following error:

<img width="612" alt="image" src="https://github.com/webdriverio/webdriverio/assets/87139317/a14e6733-9e35-4167-bea3-826256a17158">

Here I provide details of my wdio.conf.ts:

`import type { Options } from '@wdio/types'
export const config: Options.Testrunner = {

    runner: 'local',
    autoCompileOpts: {
        autoCompile: true,
        tsNodeOpts: {
            project: './tsconfig.json',
            transpileOnly: true
        }
    },
    
    port: 4723,

    specs: [
        './features/**/*.feature'
    ],
    
    exclude: [
        // 'path/to/excluded/files'
    ],

    maxInstances: 1,

    capabilities: [{
        platformName: 'Android',
        'appium:app': './apk/app-tae-release.apk',
        'appium:platformVersion': '13.0',
        'appium:deviceName': 'AndroidAPI33O',
        'appium:automationName': 'UiAutomator2',
        'appium:autoGrantPermissions': true
    }],

    logLevel: 'info',
    bail: 0,
    baseUrl: 'http://localhost',
    waitforTimeout: 10000,
    connectionRetryTimeout: 120000,
    connectionRetryCount: 3,
    services: ['appium'],
    framework: 'cucumber',
    reporters: ['spec','junit',['allure', {outputDir: 'allure-results'}],'slack'],
    cucumberOpts: {
        requireModule: ['ts-node/register'],
        require: ['./definitions/*.ts'],
        backtrace: false,
        format: ['pretty'],
        dryRun: false,
        failFast: false,
        snippets: true,
        source: true,
        strict: false,
        tagExpression: '',
        timeout: 60000,
        ignoreUndefinedDefinitions: false
    },`

TSCONFIG.JSON
`{
    "compilerOptions": {
        "moduleResolution": "node",
        "module": "commonjs",
        "target": "es2022",
        "outDir": "./dist",
        "rootDir": "./",
        "types": [
            "node",
            "@wdio/globals/types",
            "expect-webdriverio",
            "@wdio/cucumber-framework",
            "@wdio/appium-service"
        ],
        "skipLibCheck": true,
        "noEmit": true,
        "allowImportingTsExtensions": true,
        "resolveJsonModule": true,
        "isolatedModules": true,
        "strict": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "noFallthroughCasesInSwitch": true
    },
    "include": [
        "./**/*.ts"
    ]
}`

How can I resolve this error?

**Comment by christian-bromann** [comment]
Can you provide a minimal reproducible example as GitHub repository?

**Reply by bustahxc** [reply]
Sorry for the delay, here is the GitHub repository with the example that produces the error: https://github.com/bustahxc/repo-automation-3.0

I'll be waiting for your help! Thank you!

**Reply by christian-bromann** [reply]
Please make the following changes:

- in `tsconfig.json`: set `"module": "ESNext",`
- in `package.json`: add `"type": "module",`

**Reply by bustahxc** [reply]
I made the changes, but I'm still seeing the same error.
<img width="820" alt="image" src="https://github.com/webdriverio/webdriverio/assets/87139317/99244702-68cc-4357-ab62-b2326fb76ac1">
<img width="820" alt="image" src="https://github.com/webdriverio/webdriverio/assets/87139317/662f7391-7bab-41ab-8882-9d5d7a48014f">

<img width="576" alt="image" src="https://github.com/webdriverio/webdriverio/assets/87139317/3eeff7d3-9900-471e-b57a-b3170f4fcaf4">

**Reply by christian-bromann** [reply]
I forgot: please also remove `allowImportingTsExtensions` from your `tsconfig.json`

**Comment by bustahxc** [comment, marked as answer]
> I forgot: please also remove `allowImportingTsExtensions` from your `tsconfig.json`

@christian-bromann Thank you very much!! I made the mentioned changes and the error was fixed correctly.
Greetings!

**Comment by broisnischal** [comment]
@christian-bromann  I have done both them but i am still facing the issue,. 

node:internal/process/esm_loader:46
      internalBinding('errors').triggerUncaughtException(

**Reply by christian-bromann** [reply]
@neeswebservices can you provide a reproducible example?

**Comment by ert78gb** [comment]
FYI: https://github.com/TypeStrong/ts-node/issues/2026#issuecomment-2092751588

**Reply by christian-bromann** [reply]
fwiw: we are migrating to tsx
